train.py

```diff
@@ -151,17 +151,25 @@ def loss_fn(pred, labels):
         for m in model_parts:
             # apply SPMD-style PT-D techniques
             models_parallelize_fns[model_name](m, world_mesh, parallel_dims, job_config)
-            m.to_empty(device="cuda")
-            m.init_weights()
+            init_device = "cpu" if job_config.training.enable_cpu_offload else "cuda"
+            m.to_empty(device=init_device)
+            buffer_device = "cuda" if job_config.training.enable_cpu_offload else None
+            m.init_weights(buffer_device=buffer_device)
```

Comment:
Can we do one more change, to move `init_device` and `buffer_device` to be outside `if parallel_dims.pp_enabled`? This way the code can be shared by the two branches.

Also I think it reads clearer if we change them to something like
```suggestion
        if job_config.training.create_seed_checkpoint:
            init_device = "cpu"
            buffer_device = None
        elif job_config.training.enable_cpu_offload:
            init_device = "cpu"
            buffer_device = "cuda"
        else:
            init_device = "cuda"
            buffer_device = None
```